PartsApp/ExcelHelper/OperationsExcelHelper.cs

```diff
@@ -395,5 +397,53 @@ private static string xlRCtoA1(int ARow, int ACol, bool RowAbsolute = false, boo
             S = S + ARow.ToString();
             return S;
         }
+
+        public static string GetValidExcelBookTitle(string title)
+        {
+            ValidateNotNullOrEmptyTitle(title);
+
+            var forbiddenSymbols = new char[] { '/', '\\', '|', ':', '*', '?', '"', '<', '>', ',', '\'', '~', '`', '?' };
+            var validatedString = new StringBuilder();
+
+            foreach (var c in title)
+            {
+                if (forbiddenSymbols.Contains(c))
+                {
+                    validatedString.Append(' ');
+                }
+                else
+                {
+                    validatedString.Append(c);
+                }
+            }
+
+            string clearedTitle = Regex.Replace(validatedString.ToString(), @"\s+", " ");
+            ValidateNotNullOrEmptyTitle(clearedTitle);
+
+            return clearedTitle;
+        }
+
+        private static void ValidateNotNullOrEmptyTitle(string title)
+        {
+            if (string.IsNullOrWhiteSpace(title))
+            {
+                throw new ArgumentException("No operation data is provided for Excel book title");
+            }
+        }
+
+        private static string GetOperationTitle(IOperation operation)
+        {
+            if (operation is Sale sale)
+            {
+                string paymentTypeInfo = sale.PaidCash ? null : "_безнал";
+                return $"№{operation.OperationId}_{operation.OperationDate:dd-MM-yyyy}_{operation.Contragent.ContragentName}{paymentTypeInfo}";
+            }
+            else if (operation is Purchase)
+            {
+                return $"№{operation.OperationId}_{operation.OperationDate:dd-MM-yyyy}_{operation.Contragent.ContragentName}";
```

Comment:
you don't need this in two places. if 'paymentTypeInfo' is null it won't add anything to the resulting string.

Reply:
corrected